dask_jobqueue/tests/test_sge.py

```diff
@@ -107,19 +107,23 @@ def test_complex_cancel_command(loop):
     with SGECluster(
         walltime="00:02:00", cores=1, processes=1, memory="2GB", loop=loop
     ) as cluster:
-        username = "root"
-        cluster.cancel_command = "qdel -u {}".format(username)
+        with Client(cluster) as client:
+            username = "root"
+            cluster.cancel_command = "qdel -u {}".format(username)
 
-        cluster.scale(2)
+            cluster.scale(2)
 
-        start = time()
-        while not cluster.running_jobs:
-            sleep(0.100)
-            assert time() < start + QUEUE_WAIT
+            start = time()
+            while not client.scheduler_info()["workers"]:
+                sleep(0.100)
+                assert time() < start + QUEUE_WAIT
 
-        cluster.stop_all_jobs()
+            # TODO: Is there a replacement for .stop_all_jobs? stop_all_jobs
+            # does make sure that the pending jobs get qdeled.
+            # cluster.stop_all_jobs()
+            cluster.scale(0)
```

Comment:
Does this not do that?

Reply:
I think what I had in mind at the timem was that `cluster.stop_all_jobs()` is useful in itself: 
* it will `qdel` pending jobs (i.e. jobs that have been submitted but whose dask workers have not had time to connect to the dask scheduler, for example because the jobs are still sitting in the queue)
* this is useful if you want to cleanly remove all jobs that were created from the `cluster` object. Yes all the jobs will eventually be in running state and terminate after some time if the scheduler is gone (through the `death-timeout` mechanism). Sometimes it feels nicer not to have an explicit way of cleaning up.
* there may be a replacement for `cluster.stop_all_jobs()` that I am not aware of.


To be explicit: `cluster.scale(0)` will `qdel` only the jobs whose dask workers have connected to the dask scheduler.

Reply:
I think we learned that this was fine. Resolved.